src/components/AppSelector/AppSelector.tsx

```diff
@@ -1,118 +1,74 @@
-import { ReactComponent as DropDownIcon } from '../../assets/icons/ic-chevron-down.svg';
-import StateManager, { ControlProps } from 'react-select';
+import React from 'react';
+import { showError } from '../common';
+import AsyncSelect from 'react-select/async';
+import { ServerErrors } from '../../modals/commonTypes';
+import { getAppList } from '../app/service';
+import { appSelectorStyle, DropdownIndicator } from './AppSelectorUtil';
 
-const AppSelector: React.FC<{
-    primaryKey: string; //url match
-    primaryValue: string;
-    matchedKeys: string[];
-    api: () => Promise<any>;
-    apiPrimaryKey?: string; //primary key to generate map
-    onChange?: ({ label, value }) => void;
-    formatOptionLabel?: ({ label, value, ...rest }) => React.ReactNode;
-    filterOption?: (option: any, searchString: string) => boolean;
-}> = ({ primaryKey, primaryValue, matchedKeys, api, apiPrimaryKey, onChange, formatOptionLabel = null, filterOption = null }) => {
-    const [loading, result, error, reload] = useAsync(api, []);
-    const listMap = mapByKey(result?.result || [], apiPrimaryKey || primaryKey);
-    const { path } = useRouteMatch();
-    const params = useParams();
-    const { push } = useHistory();
-    function selectApp(selected) {
-        if (onChange) {
-            onChange(selected)
-            return
+interface AppSelectorType {
+    onChange: ({ label, value }) => void;
+    appId: number;
+    appName: string;
+}
+export default function AppSelector({ onChange, appId, appName }: AppSelectorType) {
+    const defaultOption = [{ value: appId, label: appName }];
+    let selectedValue = defaultOption[0];
+    let timeoutId;
+    function noOptionsMessage(inputObj: { inputValue: string }): string {
+        if (inputObj && (inputObj.inputValue === '' || inputObj.inputValue.length < 3)) {
+            return 'Type 3 chars to see matching results';
         }
-        const keys = listMap.get(selected.value);
-        const replacements = [...matchedKeys].reduce((agg, curr) => ({ ...agg, [curr]: keys[curr] }), {});
-        const newUrl = generatePath(path, { ...replacements, [primaryKey]: selected.value });
-        push(newUrl);
+        return 'No matching results';
     }
+
+    const appListOptions = (inputValue: string): Promise<[]> =>
+        new Promise((resolve) => {
+            if (timeoutId) {
+                clearTimeout(timeoutId);
+            }
+            timeoutId = setTimeout(() => {
+                if (inputValue.length < 3) {
+                    resolve([]);
+                    return;
+                }
+                getAppList({
+                    appNameSearch: inputValue,
+                    sortBy: 'appNameSort',
+                    sortOrder: 'ASC',
+                    size: 50,
+                })
+                    .then((response) => {
+                        let appList = [];
+                        if (response.result && !!response.result.appContainers) {
```

Comment:
Ref. this https://github.com/devtron-labs/dashboard/pull/322/files?file-filters%5B%5D=.tsx&show-viewed-files=true#r823407047